src/commands/cmd_tdigest.cc

```diff
@@ -238,6 +238,76 @@ class CommandTDigestRevRank : public TDigestRankCommand<true> {};
 
 class CommandTDigestRank : public TDigestRankCommand<false> {};
 
+template <bool Reverse>
+class TDigestByRankCommand : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    key_name_ = args[1];
+
+    std::set<std::string> unique_inputs_set(args.begin() + 2, args.end());
+    origin_inputs_.assign(args.begin() + 2, args.end());
+
+    unique_inputs_.reserve(unique_inputs_set.size());
+    size_t i = 0;
+    for (const auto &input : unique_inputs_set) {
+      auto value = ParseInt<int>(input);
+      if (!value) {
+        return {Status::RedisParseErr, errValueNotInteger};
+      }
+      if (*value < 0) {
+        return {Status::InvalidArgument, errInvalidRankValue};
+      }
+      unique_inputs_.push_back(*value);
+      unique_inputs_order_[input] = i;
+      ++i;
+    }
+    return Status::OK();
+  }
+  Status Execute(engine::Context &ctx, Server *srv, Connection *conn, std::string *output) override {
+    TDigest tdigest(srv->storage, conn->GetNamespace());
+    std::vector<double> result;
+    result.reserve(origin_inputs_.size());
+
+    if (const auto s =
+            [&]() {
+              if constexpr (Reverse) {
+                return tdigest.ByRevRank(ctx, key_name_, unique_inputs_, result);
+              } else {
+                return tdigest.ByRank(ctx, key_name_, unique_inputs_, result);
+              }
+            }();
+        !s.ok()) {
+      if (s.IsNotFound()) {
+        return {Status::RedisExecErr, errKeyNotFound};
+      }
+      return {Status::RedisExecErr, s.ToString()};
+    }
+
+    std::vector<std::string> ranks;
+    ranks.reserve(origin_inputs_.size());
+    auto is_resp3 = conn->GetProtocolVersion() == RESP::v3;
+    for (const auto &v : origin_inputs_) {
+      if (is_resp3) {
+        ranks.push_back(redis::Double(redis::RESP::v3, result[unique_inputs_order_[v]]));
+      } else {
+        ranks.push_back(redis::BulkString(fmt::format("{}", result[unique_inputs_order_[v]])));
+      }
+    }
```

Comment:
```suggestion
    for (const auto &v : origin_inputs_) {
      auto rank_value = result[unique_inputs_order_[v]];
      if (is_resp3) {
        ranks.push_back(redis::Double(redis::RESP::v3, rank_value));
      } else {
        ranks.push_back(redis::BulkString(fmt::format("{}", rank_value)));
      }
    }
```

Maybe this way could be cleaner.